src/core/CLucene/index/IndexWriter.cpp

```diff
@@ -1246,39 +1246,30 @@ void IndexWriter::indexCompaction(std::vector<lucene::store::Directory *> &src_d
     CND_CONDITION(dest_dirs.size() > 0, "Destination directory not found.");
     this->_trans_vec = std::move(trans_vec);
 
-    // order mapping: dir -> segment info -> segment reader
-    addIndexesSegments(src_dirs);
-
     // create segment readers
-    int32_t totDocCount = 0;
-    int numSegments = segmentInfos->size();
-    assert(numSegments > 0);
+    int numIndices = src_dirs.size();
 
     //Set of IndexReaders
     if (infoStream != NULL) {
-        message(string("src index dir size: ") + Misc::toString(numSegments));
+        message(string("src index dir size: ") + Misc::toString(numIndices));
     }
-    for (int32_t i = 0; i < numSegments; i++) {
-        SegmentInfo *si = segmentInfos->info(i);
-        IndexReader *reader = SegmentReader::get(si, MERGE_READ_BUFFER_SIZE, false /* mergeDocStores */);
+    for (int32_t i = 0; i < numIndices; i++) {
+        IndexReader* reader = lucene::index::IndexReader::open(src_dirs[i], MERGE_READ_BUFFER_SIZE, false);
         readers.push_back(reader);
-        totDocCount += reader->numDocs();
         if (infoStream != NULL) {
             message(src_dirs[i]->toString());
         }
     }
-    if (infoStream != NULL) {
-        message(string("index compaction total doc count: ") + Misc::toString(totDocCount));
-    }
+    assert(readers.size() == numIndices);
 
     // check hasProx
     bool hasProx = false;
     {
         if (!readers.empty()) {
-            auto reader = static_cast<SegmentReader*>(readers[0]);
-            hasProx = reader->getFieldInfos()->hasProx();
+            IndexReader* reader = readers[0];
+            hasProx = reader->hasProx();
```

Comment:
why from fieldinfo to reader level

Reply:
Remove this API and use `reader->getFieldInfos()->hasProx()` instead.